### DISCUSSION TITLE
What is the difference between options objective_bound and objective_target?

### DISCUSSION THREAD

**Discussion by rschwarz** [ERGO-Code/HiGHS#1127] [Q&A] [answered]
I see two options named `objective_bound` and `objective_target`, but I can't clearly identify their meaning from the descriptions alone.

In particular, I'd like to know which of the two (if any) is equivalent to Gurobi's [cutoff](https://www.gurobi.com/documentation/10.0/refman/cutoff.html), ie, pruning nodes with a dual bound worse than the given value.

I'm guessing that `objective_bound` is the better fit.

But what would `objective_target` mean, then? Is it a termination criterion for stopping when a solution at least as good as the target was found?

**Comment by jajhall** [comment, marked as answer]
When the dual simplex objective (for a dual feasible point) reaches the `objective bound` then the algorithm terminates. There is a guarantee that the optimal (primal) objective value can be no better than `objective bound`. Indeed, in general, it will be worse than this value.

The `objective_target` is an (unimplemented) option for the primal simplex solver. What it will do is stop the primal simplex solver when the primal objective (for a primal feasible point) reaches the `objective_target`. There is a guarantee that the optimal (primal) objective value is at least the `objective target`. Unlike the `objective_bound`, the `objective_target` is a value that's actually attained. Hence the different names. Gurobi appears to use `Cutoff` for both, but I prefer the distinction.

**Reply by rschwarz** [reply]
Thanks for the explanation. So in the context of MIP solving, is the value of `objective_bound` is then used for all node relaxations, effectively pruning nodes worse than this bound?

**Reply by jajhall** [reply]
In the MIP solver there's a known best dual bound (from a node solution) that's used to prune nodes that can't be better and also used to set `objective_bound` so that when nodes with a better dual bound are solved, the dual simplex solver doesn't run longer than necessary.
